image-create-custom

```diff
@@ -0,0 +1,82 @@
+#!/bin/bash
+
+parse_args() {
+    for arg in $@; do
+            if [[ "$arg" =~ ^rhel(-[[:digit:]]){2}(-([[:digit:]]{8}-[[:digit:]]|latest))$ ]]; then
+                image="$arg"
+                break
+            fi
+    done
+
+    if [ ! "$image" ]; then
+        echo "Missing or unsupported image (rhel-x-y-YYYYMMDD-B or rhel-x-y-latest)."
+        exit 1
+    else
+        echo "Using image '$image'"
+    fi
+}
+
+create_bootstrap() {
+    echo "#!/bin/bash
+
+BASE=\$(dirname \$0)
+\$BASE/virt-install-fedora \"\$1\" x86_64 \"$install_url\"" > $bootstrap_script
+    chmod +x $bootstrap_script
+}
+
+# parse script arguments and determine RHEL major and minor version
+parse_args "$@"
+major=${image:5:1}
+minor=${image:7:1}
+compose=${image#rhel-?-?-}
+compose=${compose//-/.}
+
+# determine image type and set compose ID accordingly
+if [[ "$image" =~ ^rhel-8-.-.{8}-.?$ ]]; then
+    compose="RHEL-$major.$minor.0-$compose"
+elif [[ "$image" =~ ^rhel-7-.-.{8}-.?$ ]]; then
+    compose="RHEL-$major.$minor-$compose"
+else
+    compose="latest-RHEL-$major.$minor"
+fi
+
+## don't rebuild the image by default if it already exists
+#image_path="$(dirname $0)/images/$image"
+#if [[ "$recreate" != "1" && -f $image_path ]]; then
+#    echo "Image $image already exists, use '-r' to recreate."
+#    exit 1
+#fi
+
+# determine download server URL prefix (also based on machine location)
+if [[ "$HOSTNAME" =~ brq ]]; then
+    url_prefix="http://download.eng.brq.redhat.com"
+else
+    url_prefix="http://download.eng.bos.redhat.com"
+fi
```

Comment:
Use download.devel.redhat.com for automatic location based mirror selection.

Reply:
Thanks for pointing this out, I didn't know that.